packages/api/tests/endpoints/test-granules.js

```diff
@@ -2427,429 +2581,389 @@ test.serial('PUT sets a default value of false for `published` if one is not set
-    status: 'completed',
-  };
-
-  const expressRequest = {
-    params: {
-      collectionId: t.context.collectionId,
-      granuleName: apiGranule.granuleId,
-    },
-    body: updatedGranule,
-    testContext: {
-      knex,
-      esClient: fakeEsClient,
-    },
+    ...newApiGranule,
+    updatedAt: Date.now(),
+    createdAt: Date.now(),
   };
 
-  const response = buildFakeExpressResponse();
-
-  await put(expressRequest, response);
-  t.true(response.boom.badRequest.calledWithMatch('something bad'));
+  await request(app)
+    .patch(`/granules/${newApiGranule.granuleId}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(updatedGranule)
+    .expect(200);
 
   const actualPgGranule = await t.context.granulePgModel.get(t.context.knex, {
     cumulus_id: newPgGranule.cumulus_id,
   });
+  const translatedGranule = await translatePostgresGranuleToApiGranule({
+    granulePgRecord: actualPgGranule,
+    knexOrTransaction: knex,
+  });
 
-  t.deepEqual(
-    actualPgGranule,
-    newPgGranule
-  );
-  t.deepEqual(
-    await t.context.esGranulesClient.get(
-      apiGranule.granuleId
-    ),
-    esRecord
-  );
+  const { Messages } = await sqs()
+    .receiveMessage({
+      QueueUrl: t.context.QueueUrl,
+      WaitTimeSeconds: 10,
+    })
+    .promise();
+  const snsMessageBody = JSON.parse(Messages[0].Body);
+  const publishedMessage = JSON.parse(snsMessageBody.Message);
 
-  const { Messages } = await sqs().receiveMessage({
-    QueueUrl: t.context.QueueUrl,
-    WaitTimeSeconds: 10,
-  }).promise();
-  t.is(Messages, undefined);
+  t.deepEqual(publishedMessage.record, translatedGranule);
+  t.is(publishedMessage.event, 'Update');
 });
 
-test.serial('PUT adds granule if it does not exist and returns a 201 status', async (t) => {
+test.serial(
+  "create() sets a default createdAt value for passed granule if it's not set by the user",
+  async (t) => {
+    const { esClient, executionUrl, knex } = t.context;
+
+    const { apiGranule } = await createGranuleAndFiles({
+      dbClient: knex,
+      esClient,
+      granuleParams: {
+        status: 'running',
+        execution: executionUrl,
+      },
+    });
+
+    const newGranuleId = randomId('granule');
+    const createGranuleFromApiMethodStub = sinon.stub();
+    const updatedGranule = {
+      ...apiGranule,
+      granuleId: newGranuleId,
+    };
+    delete updatedGranule.createdAt;
+    const expressRequest = {
+      params: {
+        granuleName: updatedGranule.granuleId,
+      },
+      body: updatedGranule,
+      testContext: {
+        knex,
+        createGranuleFromApiMethod: createGranuleFromApiMethodStub,
+      },
+    };
+    const response = buildFakeExpressResponse();
+    await create(expressRequest, response);
+
+    t.truthy(createGranuleFromApiMethodStub.getCalls()[0].args[0].createdAt);
+  }
+);
+
+test.serial(
+  "patch() sets a default createdAt value for new granule if it's not set by the user",
```

Comment:
Fix test title

Reply:
Don't we need to also test patch() as well then?

Reply:
https://github.com/nasa/cumulus/pull/3225/commits/5da5d848f024fcc325c64bfe30ba0449446cc2dd

I don't think we need a separate test for `patch()` in this case because `patch()` calls `patchGranule` and it's the responsibility of `patchGranule` to default the createdAt field (via `_setNewGranuleDefaults`)

Reply:
This works, particularly in light of the refactor further down/commentary.